File: trunk/csharp/UrakawaToolkit/property/channel/IChannelsPropertyFactory.cs
using System;
using System.Collections.Generic;
using System.Text;
using urakawa.property;

namespace urakawa.property.channel
{
	/// <summary>
	/// <see cref="PropertyFactory"/> that supports creation of <see cref="ChannelsProperty"/>s
	/// </summary>
	public interface IChannelsPropertyFactory : IGenericPropertyFactory
	{
		/// <summary>
		/// Creates a <see cref="ChannelsProperty"/> of default type
		/// </summary>
		/// <returns>The created <see cref="ChannelsProperty"/></returns>
		ChannelsProperty createChannelsProperty();
	}
}

File: trunk/java/core/src/org/daisy/urakawa/property/channel/IChannelsPropertyFactory.java
package org.daisy.urakawa.property.channel;

import org.daisy.urakawa.IWithPresentation;
import org.daisy.urakawa.xuk.IXukAble;
  

/**
 * <p>
 * This is the factory that creates
 * {@link org.daisy.urakawa.property.channel.IChannelsProperty} instances.
 * </p>
 * 
 * @see org.daisy.urakawa.property.PropertyFactory
 * @leafInterface see {@link org.daisy.urakawa.LeafInterface}
 * @see org.daisy.urakawa.LeafInterface
 * @stereotype OptionalLeafInterface
 * @depend - Create - org.daisy.urakawa.property.channel.IChannelsProperty
 * @depend - Aggregation 1 org.daisy.urakawa.IPresentation
 */
public interface IChannelsPropertyFactory extends IXukAble, IWithPresentation {
	/**
	 * <p>
	 * Creates a new property, not yet associated to a node.
	 * </p>
	 * <p>
	 * This factory method does not take any argument and creates an object of
	 * the default type.
	 * </p>
	 * 
	 * @return cannot be null.
	 */
	public IChannelsProperty createChannelsProperty();
}
